### PR TITLE
Arch: Fixed error on null window preset creation parameters

### PR DESCRIPTION
**Pull request by yorikvanhavre**
Fixes yorikvanhavre/BIM_Workbench#116

Thank you for creating a pull request to contribute to FreeCAD! Place an "X" in between the brackets below to "check off" to confirm that you have satisfied the requirement, or ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10) if there is something you don't understand.

- [x]  Your Pull Request meets the requirements outlined in section 5 of [CONTRIBUTING.md](https://github.com/FreeCAD/FreeCAD/blob/master/CONTRIBUTING.md) for a Valid PR

Please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [1.0 Changelog Forum Thread](https://forum.freecad.org/viewtopic.php?f=10&t=69438).

---

### CODE REVIEW

File: src/Mod/Arch/ArchWindowPresets.py [hunk 1]
```diff
@@ -51,6 +51,11 @@ def makeSketch(windowtype,width,height,h1,h2,h3,w1,w2,o1,o2):
         w2 = float(w2)
         o1 = float(o1)
         o2 = float(o2)
+        # h1, h2, w1, w2 cannot be null (for now)
+        # TODO allow these to be null (don't create the component if so)
+        if not h1 or not h2 or not w1 or not w2:
+            FreeCAD.Console.PrintError("H1, H2, W1 and W2 parameters cannot be zero. Aborting\n")
+            return
```
Comment (0penBrain) [inline src/Mod/Arch/ArchWindowPresets.py:58]: Why not something just like
```
if h1*h2*w1*w2 == 0:
    FreeCAD.Console.PrintError("H1, H2, W1 and W2 parameters cannot be zero. Aborting\n")
    return
```
at the very beginning of the function ?

Comment (yorikvanhavre) [reply src/Mod/Arch/ArchWindowPresets.py:58]: Indeed that's more elegant. But in any case, this test should vanish in the future (see the TODO comment above)

Comment (yorikvanhavre) [reply src/Mod/Arch/ArchWindowPresets.py:58]:
```suggestion
        # h1, h2, w1, w2 cannot be null (for now)
        # TODO allow these to be null (don't create the component if so)
        if h1*h2*w1*w2 == 0:
            FreeCAD.Console.PrintError("H1, H2, W1 and W2 parameters cannot be zero. Aborting\n")
            return
```

**Comment by 0penBrain** [comment]
Code is good to me (commits could be squashed, and introduced check could be done sooner in the function though).

But I won't dare to merge as I absolutely know nothing about Arch. :smile:

**Comment by yorikvanhavre** [comment]
I think we can make an exception, I'll merge this myself :)

**Pull request merged**